src/main/java/apoc/number/Number.java

```diff
@@ -0,0 +1,126 @@
+package apoc.number;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import apoc.Description;
+import apoc.result.DoubleResult;
+import apoc.result.LongResult;
+import apoc.result.StringResult;
+
+/**
+ * 
+ * @since 25.8.2016
+ * @author inserpio
+ */
+public class Number {
+
+  @Procedure
+  @Description("apoc.number.formatLong(number, pattern, locale) yield value | format a long using a (optional) pattern and for a (optional) locale to produce a string")
+  public Stream<StringResult> formatLong(final @Name("number") long number, @Name("pattern") String pattern, @Name("language") String language) {
```

Comment:
we could have one `format` method that either takes a long or double i.e. an object and checks at runtime.
also the simple variant should just take a pattern and use the locale of the server
or even simpler use a default sensible pattern
like two decimal places for doubles and thousands grouping